src/test/java/guitests/guihandles/GuiHandle.java

```diff
@@ -28,15 +33,39 @@ public GuiHandle(GuiRobot guiRobot, Stage primaryStage) {
     public <T> T as(Class<? extends GuiHandle> clazz) {
         try {
             Constructor<?> ctor = clazz.getConstructor(GuiRobot.class, Stage.class);
-            Object object = ctor.newInstance(new Object[] { guiRobot, primaryStage });
+            Object object = ctor.newInstance(new Object[] { guiRobot, primaryStage});
             return (T)object;
         } catch (Exception e) {
             throw new RuntimeException("Cannot create gui handle of type " + clazz.getName(), e);
         }
     }
 
+    public void focusOnWindow(String stageTitle) {
+        LoggerManager.getLogger(this.getClass()).info("Focusing " + stageTitle);
+        java.util.Optional<Window> window = guiRobot.listTargetWindows()
+                .stream()
+                .filter(w
+                        -> w instanceof Stage
+                        && ((Stage)w).getTitle().equals(stageTitle)).findAny();
+
+        if(!window.isPresent()) {
+            LoggerManager.getLogger(this.getClass()).fatal("Can't find stage " + stageTitle + ", Therefore, aborting focusing");
+            return;
+        }
+
+        guiRobot.targetWindow(window.get());
+        guiRobot.interact(() -> window.get().requestFocus());
+        LoggerManager.getLogger(this.getClass()).info("Finishing focus " + stageTitle);
+    }
```

Comment:
Looks like this should be a method of `mainGui` (as it involves all windows shown at that moment)?